onnxruntime/core/session/ep_api.cc

```diff
@@ -34,6 +37,18 @@ ORT_API_STATUS_IMPL(CreateEpDevice, _In_ OrtEpFactory* ep_factory,
     ep_device->ep_metadata = *ep_metadata;
   }
 
+  if (ep_device->ep_metadata.Entries().find(kOrtEpDevice_EpMetadataKey_Version) !=
+      ep_device->ep_metadata.Entries().end()) {
+    return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT,
+                                 "The provided EP metadata should not explicitly specify the EP version.");
+  }
+
+  {
+    std::string ep_version = ep_factory->GetVersion(ep_factory);
```

Comment:
Should this check the ORT version the EP was compiled against before calling the new function (ort_version_supported)?